nexus/db-queries/src/db/datastore/inventory.rs

```diff
@@ -1537,17 +1589,29 @@ impl DataStore {
         // Fetch records of RoT pages found.
         let inv_rot_page_rows = {
             use db::schema::inv_root_of_trust_page::dsl;
-            dsl::inv_root_of_trust_page
+
+            let mut rot_pages = Vec::new();
+
+            let mut paginator = Paginator::new(batch_size);
+            while let Some(p) = paginator.next() {
+                let mut batch = paginated(
+                    dsl::inv_root_of_trust_page,
+                    dsl::time_collected,
+                    &p.current_pagparams(),
+                )
                 .filter(dsl::inv_collection_id.eq(id))
-                .limit(sql_limit)
                 .select(InvRotPage::as_select())
                 .load_async(&*conn)
                 .await
                 .map_err(|e| {
                     public_error_from_diesel(e, ErrorHandler::Server)
-                })?
+                })?;
+                paginator = p.found_batch(&batch, &|row| row.time_collected);
```

Comment:
Same -- and this one's also multicolumn I think.  (When in doubt for what should go here, check out the unique indexes / primary keys in schema/crdb/dbinit.sql.)